client/service/src/task_manager/mod.rs

```diff
@@ -127,7 +124,20 @@ impl SpawnTaskHandle {
 			}
 		};
 
-		let join_handle = self.executor.spawn(Box::pin(future.in_current_span()), task_type);
+		let future: Pin<Box<dyn Future<Output = ()> + Send>> =
```

Comment:
Wasn't there a `std::boxed::PinBox` at some point? Maybe I'm wrong.

Reply:
:woman_shrugging: it doesn't ring a bell

Reply:
@dvdplm I assume you mean this: https://docs.rs/futures/0.3.12/futures/future/type.BoxFuture.html

Reply:
ah that's the one. Can we use it here @cecton?

Reply:
Done in d3504f45b

Sorry I used that before and I completely forgot about it somehow XD